src/builder-cache.c

```diff
@@ -418,7 +418,7 @@ builder_cache_lookup (BuilderCache *self,
       g_variant_get (variant, "(a{sv}aya(say)&s&stayay)", NULL, NULL, NULL,
                      &subject, NULL, NULL, NULL, NULL);
 
-      if (strcmp (subject, self->current_checksum) == 0)
+      if (subject != NULL && strcmp (subject, self->current_checksum) == 0)
```

Comment:
```suggestion
      if (g_strcmp0 (subject, self->current_checksum) == 0)
```

Maybe slightly cleaner?

Reply:
good catch. It's got a small semantic gap, though, in that it will also let both pointers being NULL pass.